### PR TITLE
[MSHARED-1347] Include invoke dynamic bootstrap types in analysis

### PR DESCRIPTION
**Pull request by vbreivik**
After this change the test case added to the issue [MDEP-576](https://issues.apache.org/jira/browse/MDEP-576): [dependency-plugin-vs-java-lambda](https://github.com/vdergachev/dependency-plugin-vs-java-lambda) passes OK. 

The above test is reproduced in a smaller scale by creating two classes in their own packages and analyse the one package.  

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes. Also be sure having selected the correct component.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
@vbreivik please fix formatting issues

**Comment by vbreivik** [comment]
> @vbreivik please fix formatting issues

@slawekjaranowski done. 

Guess spotless tricked me, it does not run on java 8. 
Switched to 17 and ran spotless:apply and pushed

File: src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DefaultMethodVisitor.java [hunk 1]
```diff
@@ -157,4 +160,14 @@ private void addTypeSignature(final String signature) {
             new SignatureReader(signature).acceptType(signatureVisitor);
         }
     }
+
+    @Override
+    public void visitInvokeDynamicInsn(
```
Comment (elharo) [inline src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DefaultMethodVisitor.java:165]: Insn --> Instruction
I had to think about this one to figure it out.
In general, avoid abbreviations

File: src/test/java/org/apache/maven/shared/dependency/analyzer/asm/ASMDependencyAnalyzerTest.java [hunk 2]
```diff
@@ -40,4 +43,12 @@ void test() throws Exception {
 
         assertThat(result).isNotEmpty();
     }
+
+    @Test
+    void verify_implicit_class_reference_included_in_used_classes() throws IOException {
```
Comment (elharo) [inline src/test/java/org/apache/maven/shared/dependency/analyzer/asm/ASMDependencyAnalyzerTest.java:48]: naming convention seems wonky here

File: src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DefaultMethodVisitor.java (hunk 1 above)
Comment (vbreivik) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DefaultMethodVisitor.java:165]: The naming is controlled by ASM. This class extends MethodVisitor from ASM.

File: src/test/java/org/apache/maven/shared/dependency/analyzer/asm/ASMDependencyAnalyzerTest.java (hunk 2 above)
Comment (vbreivik) [reply src/test/java/org/apache/maven/shared/dependency/analyzer/asm/ASMDependencyAnalyzerTest.java:48]: Snake case is a habit, I find it more readable. I have no strong opinions. If it should be changed to camelCase then let me know and I will update.

Please also check if my logic is sound. I am no expert in ASM and byte code. Naming is easily fixed.

**Review by elharo — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #158

**Pull request merged**